src/modules/navigator/mission.cpp

```diff
@@ -196,18 +191,21 @@ Mission::on_activation()
 		updateCachedItemsUpToIndex(_current_mission_index - 1);
 	}
 
-	unsigned resume_index;
+	uint16_t resume_index;
 
 	if (_inactivation_index > 0 && cameraWasTriggering()
 	    && getPreviousPositionItemIndex(_mission, _inactivation_index - 1, resume_index)) {
 		// The mission we are resuming had camera triggering enabled. In order to not lose any images
 		// we restart the mission at the previous position item.
 		// We will replay the cached commands once we reach the previous position item and have yaw aligned.
+
+		checkClimbRequired(resume_index);
 		set_current_mission_index(resume_index);
 
 		_align_heading_necessary = true;
 
 	} else {
+		checkClimbRequired(resume_index);
```

Comment:
At this point `resume_index` might be uninitialized no?

`uint16_t resume_index;` and `getPreviousPositionItemIndex(_mission, _inactivation_index - 1, resume_index)` might have returned false so we don't know what the value is on this line.

Reply:
Very good point, addressed it in a7489ff3ce7681e4f09d7f880669ec3bce805dc2